tests/test_stacks.py

```diff
@@ -670,3 +671,20 @@ def test_integrated_stack_has_revision_security_status(
     revisions = {r["phid"]: r for r in response.json["revisions"]}
     assert not revisions[public_revision["phid"]]["is_secure"]
     assert revisions[secure_revision["phid"]]["is_secure"]
+
+
+def test_revisionstack():
```

Comment:
How does `iter_stack_from_base` behave if there is more than one node in `stack.base_revisions()`? E.g., if a stack of 3 revisions has a revision with two predecessors. Would be good to add a test for this, and maybe document it in the docstring as well.

Reply:
As of right now it simply picks the first available base revision and walks the stack from there. I will document this in the docstring. I suppose I could alternatively disallow any attempt to create uplift revisions where there is a non-linear stack path, just to keep things simple.

I had to change the implementation to work in this way because the `landing_path` variable is pulled from `calculate_landable_subgraphs`, where closed revisions are counted as "not landable". This is a problem because we want to be able to uplift closed revisions.